crates/driver/src/infra/api/mod.rs

```diff
@@ -69,6 +70,16 @@ impl Api {
             let router = routes::solve(router);
             let router = routes::reveal(router);
             let router = routes::settle(router);
+
+            let mut bad_tokens =
+                bad_tokens::Detector::new(solver.bad_token_detection().tokens_supported.clone());
+            if solver
+                .bad_token_detection()
+                .enable_simulation_based_bad_token_detection
+            {
+                bad_tokens.with_simulation_detector(self.bad_token_detector.clone());
+            }
```

Comment:
Still not sure how this is expected to be deployed. At some point in time, will there be 3 bad token detectors enabled? orderbook+autopilot+this one.

Reply:
There will be a transition period where we have the bad token detectors running on the protocol level (autopilot/orderbook) and the driver.
That is mostly intended to see if the driver implementation has reasonable performance and doesn't flag tokens incorrectly.
Then we'd disable the bad token detection in the protocol level and see if things are stable. I expect the initial implementation will probably not cover all the cases so we'll get some alerts. Then we'll iron out the implementation to have it as smooth as it's right now.
Finally we can remove the bad token detection altogether from the autopilot / orderbook.